app/main.py

```diff
@@ -45,14 +45,10 @@ def __eq__(self, other: int | float | Distance) -> bool:
         return self.km == other
 
     def __le__(self, other: int | float | Distance) -> bool:
-        if isinstance(other, Distance):
-            return not self.km > other.km
-        return not self.km > other
+        return not Distance.__gt__(self, other)
```

Comment:
magic method `__gt__` is called by `>` symbol. And also we call only classmethods through class names. So the correct solution is:
`return not self > other`